### PR TITLE
Internal: Nested elements POC [Without repeater] [ED-2256]

### PR DESCRIPTION
**Pull request by iNewLegend**

### CODE REVIEW

**Review by matipojo — CHANGES_REQUESTED** [review]

File: assets/dev/js/editor/document/elements/component.js [hunk 1]
```diff
@@ -32,6 +32,11 @@ export default class Component extends ComponentBase {
 					return false;
 				}
 
+				if ( draggedElType === parentElType ) {
+					// Support nested elements.
+					return !! elementor.widgetsCache[ parentModel.get( 'widgetType' ) ]?.support_repeater_elements;
```
[the remark this answers is not in the document]
Comment (iNewLegend) [reply assets/dev/js/editor/document/elements/component.js]: Handled.

File: assets/dev/js/editor/elements/models/element-repeater.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
+module.exports = class ElementRepeaterModel extends elementor.modules.elements.models.Element {
+	initialize( options ) {
+		if ( $e.commands.isCurrentFirstTrace( 'document/elements/create' ) ) {
+			setTimeout( () => {
+				const name = elementor.widgetsCache[ this.get( 'widgetType' ) ].name;
+
+				this.attributes.settings.get( name ).forEach( () => {
+					const container = elementor.getContainer( this.id );
+
+					const sectionView = container.view.addElement( new ElementRepeaterModel( {
```
[the remark this answers is not in the document]
Comment (iNewLegend) [reply assets/dev/js/editor/elements/models/element-repeater.js]: Handled.

File: assets/dev/js/editor/elements/views/base.js [hunk 3]
```diff
@@ -88,12 +88,29 @@ BaseElementView = BaseContainer.extend( {
...
+				ChildView = require( 'elementor-elements/views/section' );
+				break;
+
+			case 'column':
+				ChildView = require( 'elementor-elements/views/column' );
+				break;
+
+			case 'widget':
+				if ( elementor.widgetsCache[ model.get( 'widgetType' ) ]?.support_repeater_elements ) {
+					ChildView = require( 'elementor-elements/views/widget-repeater' );
```
[the remark this answers is not in the document]
Comment (iNewLegend) [reply assets/dev/js/editor/elements/views/base.js]: Handled.

File: assets/dev/js/editor/utils/helpers.js [hunk 4]
```diff
@@ -34,7 +34,9 @@ module.exports = {
 		document: {
 			section: {
 				column: {
-					widget: null,
+					widget: {
+						widget: null,
+					},
```
Comment (iNewLegend) [inline assets/dev/js/editor/utils/helpers.js:39]: Check is required.

File: assets/dev/js/editor/elements/collections/elements.js [hunk 5]
```diff
@@ -1,8 +1,11 @@
+import WidgetContainerModel from 'elementor-elements/models/widget-container-model';
+
 var ElementModel = require( 'elementor-elements/models/element' );
 
 var ElementsCollection = Backbone.Collection.extend( {
 	add: function( models, options, isCorrectSet ) {
 		if ( ( ! options || ! options.silent ) && ! isCorrectSet ) {
+			debugger;
```
Comment (iNewLegend) [inline assets/dev/js/editor/elements/collections/elements.js:8]:
```suggestion
```

File: includes/widgets/tabs-v2.php [hunk 6]
```diff
@@ -0,0 +1,452 @@
...
+				'class': [ 'elementor-tab-content', 'elementor-clearfix', 'elementor-repeater-item-' + item._id ],
+				'data-tab': tabCount,
+				'role' : 'tabpanel',
+				'aria-labelledby' : 'elementor-tab-title-' + elementUid + tabCount
+				} );
+				#>
+				<div class="elementor-tab-title elementor-tab-mobile-title" data-tab="{{ tabCount }}" role="tab">{{{ item.tab_title }}}</div>
+				<div {{{ view.getRenderAttributeString( tabContentKey ) }}}>
+					<div class="<?php echo esc_html( $this->get_children_placeholder_class() ); ?>" data-index="{{ index }}">
+					</div>
```
Comment (iNewLegend) [inline includes/widgets/tabs-v2.php:444]:
```suggestion
				<# view.printChildren( index ) #>
```

File: includes/base/widget-container-base.php [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+}
+
+abstract class Widget_Container_Base extends Widget_Base {
+
+	abstract protected function get_default_children();
+
+	protected function get_children_placeholder_class() {
+		// Empty string, means will be added at the end.
+		return '';
+	}
```
Comment (iNewLegend) [inline includes/base/widget-container-base.php:18]:
```suggestion
```
Always render into `.children` class.

**Pull request closed without merging**